index.js

```diff
@@ -1,89 +1,91 @@
+    },
 
-  /**
-   * @param {SQLStatement|string} statement
-   * @returns {this}
-   */
-  append(statement) {
-    if (statement instanceof SQLStatement) {
-      this.strings[this.strings.length - 1] += statement.strings[0]
-      this.strings.push.apply(this.strings, statement.strings.slice(1))
-      const list = this.values || this.bind
-      list.push.apply(list, statement.values)
-    } else {
-      this.strings[this.strings.length - 1] += statement
-    }
-    return this
-  }
+    /** Returns the SQL Statement for mysql */
+    get sql() {
+      return strings.join('?')
+    },
 
-  /**
-   * Use a prepared statement with Sequelize.
-   * Makes `query` return a query with `$n` syntax instead of `?`  and switches the `values` key name to `bind`
-   * @param {boolean} [value=true] value If omitted, defaults to `true`
-   * @returns this
-   */
-  useBind(value) {
-    if (value === undefined) {
-      value = true
-    }
-    if (value && !this.bind) {
-      this.bind = this.values
-      delete this.values
-    } else if (!value && this.bind) {
-      this.values = this.bind
-      delete this.bind
-    }
-    return this
-  }
+    /** Returns the SQL Statement for node-postgres */
+    get text() {
+      return strings.reduce((p, c, i) => `${p}$${i}${c}`)
+    },
 
-  /**
-   * @param {string} name
-   * @returns {this}
-   */
-  setName(name) {
-    this.name = name
-    return this
-  }
-}
+    /** Returns values for others */
+    get values() {
+      return this.bound ? void 0 : values
+    },
 
-/** Returns the SQL Statement for mysql */
-Object.defineProperty(SQLStatement.prototype, 'sql', {
-  enumerable: true,
-  get() {
-    return this.strings.join('?')
-  },
-})
+    // Methods
 
-/**
- * @param {string[]} strings
- * @param {...any} values
- * @returns {SQLStatement}
- */
-function SQL(strings) {
-  return new SQLStatement(strings.slice(0), Array.from(arguments).slice(1))
+    /**
+     * @param {SQLStatement|string} statement
+     * @returns {this}
+     */
+    append(statement) {
+      if (typeof statement === 'string') {
+        strings[strings.length - 1] += statement
+      } else {
+        strings[strings.length - 1] += statement.strings[0]
+        strings.push(...statement.strings.slice(1))
+        values.push(...statement.values)
+      }
+      return this
+    },
+
+    /**
+     * @param {string} name
+     * @returns {this}
+     */
+    setName(name) {
+      this.name = name
+      return this
+    },
+
+    /**
+     * Use a prepared statement with Sequelize.
+     * Makes `query` return a query with `$n` syntax instead of `?` and
+     * uses `bind` instead of `values`
+     * @param {boolean} [bound=true] bound If omitted, defaults to `true`
+     * @returns this
+     */
+    useBind(bound = true) {
+      this.bound = bound
+      return this
+    },
+
+    // Properties
+
+    /** Default bound value as false */
+    bound: false,
+
+    /** Default name as empty string */
+    name: '',
```

Comment:
Shouldn't a statement be not named at all by default (`undefined`)?

Reply:
same as https://github.com/felixfbecker/node-sql-template-strings/pull/103#discussion_r260771148

this whole refactoring is to have specialized, optimized, V8 internal class with an object that never deletes, or change, its fields.

Reply:
I am just afraid of breaking a DB driver that may check for a name like this:

```js
if (typeof options.name === 'string')
if (options.name !== undefined)
```

Reply:
that driver will fail when the object changes at runtime though, right? 'cause if it has a different usage for when the name is there or not, mutating the object won't solve the root cause of possible issues.

Reply:
also, as stated in here https://github.com/felixfbecker/node-sql-template-strings/blob/master/index.d.ts#L24-L27 the `name` is a string by specs/API/definition

Reply:
No, cause you're only changing the object before you pass it to the driver.
The `name` TS declaration was written before TS had support for optional class properties, it was `undefined` at runtime if not set I think.

Reply:
this is similar issue we have with other fields ... if you have `name in`